### PR TITLE
Add Order Instructions to APID

### PR DESCRIPTION
**Pull request by nitro-neal**
Adding order instructions to our apid that is conforming to :
https://github.com/TBD54566975/tbdex/pull/355 and https://github.com/TBD54566975/tbdex/issues/335

rust code for `Order Instructions` ✅ 
binding code `Order Instructions` ✅ 
bound/kt `Order Instructions` ✅ 
examples pfi and wallet updated ✅

### CODE REVIEW

**Comment by nitro-neal** [comment]
an open question: 

does there need to be something added to the http client like `get_order_instructions()`?

or would that come back in a new field in the response in submit_order()?

**Comment by KendallWeihe** [comment]
> does there need to be something added to the http client like get_order_instructions()?
> 
> or would that come back in a new field in the response in submit_order()?

No but it does need to be added to the `ReplyToMessage` enumeration set (this is tricky and we haven't talked it through before). The `OrderInstructions` originate via the PFI so the client would receive the message via either a call to `get_exchange()` or via a webhook callback (hence the `ReplyToMessage` addition).

File: docs/API_DESIGN.md [hunk 1]
```diff
@@ -376,30 +379,51 @@ CLASS QuoteDetails
...
+  METHOD sign(bearer_did: BearerDid): Error?
+  METHOD to_json_string(): string
 ```
 
-## `Order`
+## `OrderInstructions`
 
 ```pseudocode!
-CLASS Order IMPLEMENTS Message, WalletUpdateMessage
+CLASS OrderInstructions IMPLEMENTS Message
```
Comment (KendallWeihe) [inline docs/API_DESIGN.md:400]:
```suggestion
CLASS OrderInstructions IMPLEMENTS Message, ReplyToMessage
```

File: bound/kt/src/main/kotlin/tbdex/sdk/messages/OrderInstructions.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import tbdex.sdk.web5.BearerDid
+
+import tbdex.sdk.rust.OrderInstructionsDataData as RustCoreOrderInstructionsData
+typealias OrderInstructionsData = RustCoreOrderInstructionsData
+
+class OrderInstructions private constructor(
+    val metadata: MessageMetadata,
+    val data: OrderInstructionsData,
+    val signature: String,
+    internal val rustCoreOrderInstructions: tbdex.sdk.rust.OrderInstructions
```
Comment (KendallWeihe) [inline bound/kt/src/main/kotlin/tbdex/sdk/messages/OrderInstructions.kt:14]:
```suggestion
    internal val rustCoreOrderInstructions: RustCoreOrderInstructionsData
```

**Review by KendallWeihe — APPROVED** [review]
🏆 ✅ 🎉 

AMAZING work!

**Pull request merged**